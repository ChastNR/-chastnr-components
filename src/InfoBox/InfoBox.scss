.ib {
  margin-bottom: 8px;
  border-radius: 4px;
  position: relative;
  z-index: 0;
  width: 100%;
  display: flex;
  background-color: #fff;
  align-items: stretch;
  color: rgba(0, 0, 0, 0.8);
  cursor: pointer;
  user-select: none;
  border: 1px solid transparent;
  -webkit-appearance: none;
  max-width: 400px;
  -webkit-tap-highlight-color: transparent;

  &:before {
    content: "";
    position: absolute;
    top: -1px;
    right: -1px;
    bottom: -1px;
    left: -1px;
    border: 1px solid #eaecee;
    border-radius: inherit;
    background-color: #fff;
    transition: background-color 0.2s ease;
  }

  & > div {
    display: flex;
    flex-grow: 1;
    align-items: flex-start;
    border-radius: inherit;
    align-items: center;
    padding: 16px;
    margin: -1px;
  }

  .ib__stn {
    position: relative;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: stretch;
    flex: 0 0 auto;
    color: rgba(0, 0, 0, 0.8);
    min-height: 48px;
    margin-left: 16px;

    &:first-child {
      margin: 0;
    }

    &:last-child {
      align-items: flex-end;
      margin-left: auto;
    }

    & > img {
      display: inline-block;
      width: 48px;
      height: 48px;
    }

    & > span {
      font-size: 15px;
      line-height: 20px;
      letter-spacing: 0;
      color: #1771e6;
    }

    .ib__stn__title {
      margin: 1px 0;
      font-size: 15px;
      line-height: 24px;
      margin: -1px 0;
      color: rgba(0, 0, 0, 0.8);
    }

    .ib__stn__cnt {
      margin: 1px 0;
      word-wrap: break-word;
      font-size: 13px;
      line-height: 20px;
      font-weight: 400;
      letter-spacing: 0;
      margin: -1px 0;
      color: rgba(0, 0, 0, 0.54);
    }
  }
}
