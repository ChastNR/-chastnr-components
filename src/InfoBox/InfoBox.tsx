/* eslint-disable jsx-a11y/label-has-associated-control */
import classNames from 'clsx';

import Checkbox from '../Checkbox';

import './InfoBox.scss';

interface InfoBoxProps {
  additionalInfo?: React.ReactNode;
  checked?: boolean;
  className?: string;
  disabled?: boolean;
  img?: string;
  onChange?: React.ChangeEventHandler<HTMLInputElement>;
  small?: boolean;
  title: React.ReactNode;
}

const InfoBox: React.FC<InfoBoxProps> = ({
  additionalInfo,
  checked,
  children,
  className,
  disabled,
  img,
  onChange,
  small,
  title,
}) => {
  const infoBoxClassNames = classNames('ib', className, { small });

  const isImgAvailable = img && !small;

  return (
    <label className={infoBoxClassNames}>
      <div>
        <div className="ib__stn">
          <Checkbox checked={checked} disabled={disabled} onChange={onChange} />
        </div>
        {isImgAvailable && (
          <div className="ib__stn">
            <img alt="icon" src={img} />
          </div>
        )}
        <div className="ib__stn">
          <div className="ib__stn__title">{title}</div>
          <div className="ib__stn__cnt">{children}</div>
        </div>
        <div className="ib__stn">
          <span>{additionalInfo}</span>
        </div>
      </div>
    </label>
  );
};

export default InfoBox;
