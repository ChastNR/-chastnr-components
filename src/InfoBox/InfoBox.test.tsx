import { render } from '@testing-library/react';
import { InfoBox } from './InfoBox';

describe('<InfoBox />', () => {
  it('should match snapshot', () => {
    const infoBox = render(
      <InfoBox
        additionalInfo="Additional info"
        checked
        className="testClassName"
        disabled
        title="testTitle"
      >
        Text content
      </InfoBox>
    );

    expect(infoBox.container.firstChild).toMatchSnapshot();
  });

  it('image should be available', () => {
    const infoBox = render(
      <InfoBox
        additionalInfo="Additional info"
        checked
        img="testImgSrc"
        className="testClassName"
        disabled
        title="testTitle"
      >
        Text content
      </InfoBox>
    );

    const imgContainer = infoBox.container.querySelector('.ib__stn');

    expect(imgContainer).toBeDefined();
  });
});
