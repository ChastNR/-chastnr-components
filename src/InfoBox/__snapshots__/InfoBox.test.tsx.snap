// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InfoBox /> should match snapshot 1`] = `
<label
  class="ib testClassName"
>
  <div>
    <div
      class="ib__stn"
    >
      <div
        class="cb"
      >
        <input
          checked=""
          class="cb__input"
          disabled=""
          type="checkbox"
        />
        <span>
          <span>
            <span>
              <span>
                content
              </span>
            </span>
          </span>
        </span>
      </div>
    </div>
    <div
      class="ib__stn"
    >
      <div
        class="ib__stn__title"
      >
        testTitle
      </div>
      <div
        class="ib__stn__cnt"
      >
        Text content
      </div>
    </div>
    <div
      class="ib__stn"
    >
      <span>
        Additional info
      </span>
    </div>
  </div>
</label>
`;
